cmd/podman/machine/start.go

```diff
@@ -79,7 +84,35 @@ func start(_ *cobra.Command, args []string) error {
 	if err := vm.Start(vmName, startOpts); err != nil {
 		return err
 	}
+	if err := changeDockerContext(); err != nil {
+		fmt.Printf("Failed to change docker context to default. Error: %v\n", err)
+	} else {
+		fmt.Println("Docker context has been set to default")
+	}
 	fmt.Printf("Machine %q started successfully\n", vmName)
 	newMachineEvent(events.Start, events.Event{Name: vmName})
 	return nil
 }
+
+func changeDockerContext() error {
+	if runtime.GOOS != "darwin" {
+		return nil
+	}
+	pathList := strings.Split(os.Getenv("PATH"), ":")
+	var dockerBinaryPath string
+	for _, path := range pathList {
+		dockerPath := filepath.Join(path, "docker")
+		if info, err := os.Stat(dockerPath); err == nil && info.Mode().IsRegular() {
+			dockerBinaryPath = dockerPath
+			break
+		}
+	}
+	if dockerBinaryPath == "" {
+		return fmt.Errorf("docker binary not found")
+	}
+
+	cmd := exec.Command(dockerBinaryPath, "context", "use", "default")
+
+	// Run the command
+	return cmd.Run()
```

Comment:
This will return a absolutely useless error like `exit status 1` when something goes wrong.
At the very least we should capture stderr from the output, I suggest you use `CombinedOutput()` and include the output in the error message.

Reply:
Thanks, changed it.